server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -159,12 +186,22 @@ public ClusterMetadataManifest writeFullMetadata(ClusterState clusterState, Stri
             return null;
         }
 
+        // TODO: we can upload global metadata and index metadata in parallel. [issue: #10645]
```

Comment:
lets link it in META issue - https://github.com/opensearch-project/OpenSearch/issues/10523